.golangci.yml

```diff
@@ -1,209 +1,230 @@
+            - pkg: io/ioutil
+              desc: 'replaced by io and os packages since Go 1.16: https://tip.golang.org/doc/go1.16#ioutil'
+            - pkg: github.com/ghodss/yaml
+              desc: use sigs.k8s.io/yaml instead, to be consistent
+    errcheck:
+      exclude-functions:
+        - (*github.com/tektoncd/pipeline/vendor/go.uber.org/zap.SugaredLogger).Sync
+        - flag.Set
+        - os.Setenv
+        - logger.Sync
+        - fmt.Fprintf
+        - fmt.Fprintln
+        - (io.Closer).Close
+        - updateConfigMap
+    gomodguard:
+      blocked:
+        modules:
+          - github.com/ghodss/yaml:
+              recommendations:
+                - sigs.k8s.io/yaml
+    gosec:
+      excludes:
+        - G601
+  exclusions:
+    generated: lax
+    presets:
+      - common-false-positives
+      - legacy
+      - std-error-handling
+    rules:
+      - linters:
+          - forbidigo
+        path: main\.go
+      - linters:
+          - dogsled
+          - errcheck
+          - goconst
+          - gosec
+          - ineffassign
+          - maintidx
+        path: _test\.go
+      - linters:
+          - unused
+        path: test/pipelinerun_test\.go
+      - linters:
+          - containedctx
+        path: pkg/apis/config/feature_flags_test\.go
+      - linters:
+          - containedctx
+        path: pkg/pipelinerunmetrics/injection\.go
+      - linters:
+          - maintidx
+        path: pkg/pod/pod\.go
+      - linters:
+          - containedctx
+        path: pkg/pod/creds_init_test\.go
+      - linters:
+          - maintidx
+        path: pkg/reconciler/pipelinerun/pipelinerun\.go
+      - linters:
+          - containedctx
+        path: pkg/taskrunmetrics/injection\.go
+      - linters:
+          - containedctx
+        path: test/controller\.go
+      - linters:
+          - musttag
+        path: internal/sidecarlogresults/sidecarlogresults\.go
+      - linters:
+          - errchkjson
+        path: internal/sidecarlogresults/sidecarlogresults_test\.go
+      - linters:
+          - musttag
+        path: pkg/apis/pipeline/v1.*/param_types\.go
+      - linters:
+          - contextcheck
+        path: pkg/resolution/resolver/framework/testing/fakecontroller\.go
+      - linters:
+          - contextcheck
+        path: pkg/pipelinerunmetrics/metrics\.go
+      - linters:
+          - contextcheck
+        path: pkg/reconciler/pipelinerun/pipelinerun\.go
+    paths:
+      - .*/zz_generated.deepcopy.go
+      - pkg/apis/pipeline/v1beta1/openapi_generated.go
+      - vendor
+      - pkg/client
+      - pkg/spire/test
+      - third_party$
+      - builtin$
+      - examples$
+issues:
   max-issues-per-linter: 0
   max-same-issues: 0
-  include:
-  # Enable off-by-default rules for revive requiring that all exported elements have a properly formatted comment.
-  - EXC0012 # https://golangci-lint.run/usage/false-positives/#exc0012
-  - EXC0014 # https://golangci-lint.run/usage/false-positives/#exc0014
```

Comment:
should we still keep these rules, they are good for documentation

```
- EXC0012  # Checks if exported elements have comments
- EXC0014  # Checks if comments on exported elements have proper form
```

Reply:
Ah I ran `golangci-lint migrate` and it removed it, I wonder if it's still something configurable or not.

Reply:
I think we can add it later, it's not a blocker

Reply:
TODO ^

Reply:
I think golangci-lint v2 no longer has "off-by-default" functionality. 
Apart from path specific exclusions, the new configuation is

```
  exclusions:
    generated: lax
    presets:
      - common-false-positives
      - legacy
      - std-error-handling
 ```

Which I think means that those comment rules are on.
It should be possible to verify by breaking a comment and running lint

Reply:
Thank you for checking this Andrea